testng-core-api/src/main/java/org/testng/ITestNGMethod.java

```diff
@@ -220,6 +220,10 @@ default boolean hasAfterGroupsConfiguration() {
 
   void setInterceptedPriority(int priority);
 
+  int getClassHierarchyPriority();
+
+  void setClassHierarchyPriority(int priority);
```

Comment:
I dont see this method being called anywhere apart from `org.testng.internal.MethodInheritance#fixMethodInheritance`. Can you please help me understand a bit more on this intent and when does this get invoked. Also it would be good if we could include a test that ensures that this change is also being vetted out.

Reply:
That is correct, the only place where this is needed is in this org.testng.internal.MethodInheritance#fixMethodInheritance.

Later it will be used in the sorting comparators to try sort them in a class hierarchy order.  But of course, there is no guaranty that the methods will be sorted in the class hierarchy order if users have used dependsOnMethod or dependsOnGroups which have higher priority when it comes to the sorting.

Reply:
@krmahadevan IMO, these methods, like `*InterceptedPriority`,  should not be in the public API because they open a part of the implementation.

@martinaldrin What if you add the 2 methods into a dedicated interface and use the method only when the object implements the methods?

Reply:
I agree that it would be nice to hide the methods, I can have a try, but the code might be a bit more complicated if I should hide these two methods. and will require casting on several places.

Reply:
I have now moved this to an internal interface.